src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -33,33 +34,54 @@ public class HttpClientHandlerTest
         public readonly static object[][] PostServers = HttpTestServers.PostServers;
-        public readonly static object[][] DeleteServers = HttpTestServers.DeleteServers;
-
-        private static bool JsonMessageContainsKeyValue(string message, string key, string value)
-        {
-            // TODO: Align with the rest of tests w.r.t response parsing once the test server is finalized.
-            // Currently not adding any new dependencies
-            string pattern = string.Format(@"""{0}"": ""{1}""", key, value);
-            return message.Contains(pattern);
-        }
-
-        private static bool JsonMessageContainsKey(string message, string key)
-        {
-            // TODO: Align with the rest of tests w.r.t response parsing once the test server is finalized.
-            // Currently not adding any new dependencies
-            string pattern = string.Format(@"""{0}"": """, key);
-            return message.Contains(pattern);
-        }
-
+        public readonly static object[][] HttpMethods = 
+            new object[][]
+                {
+                    // Standard HTTP methods defined in RFC7231: http://tools.ietf.org/html/rfc7231#section-4.3
+                    new object[] { "GET", false },
+                    new object[] { "GET", true },
+                    new object[] { "HEAD", false },
+                    new object[] { "HEAD", true },
+                    new object[] { "POST", false },
+                    new object[] { "POST", true },
+                    new object[] { "PUT", false },
+                    new object[] { "PUT", true },
+                    new object[] { "DELETE", false },
+                    new object[] { "DELETE", true },
+                    new object[] { "OPTIONS", false },
+                    new object[] { "OPTIONS", true },
+                    new object[] { "TRACE", false },
+                    new object[] { "TRACE", true },
+                    
+                    // Custom HTTP method
+                    new object[] { "CUSTOM1", false },
+                    new object[] { "CUSTOM1", true }
+                };
+        public readonly static object[][] HttpMethodsThatAllowContent = 
+            new object[][]
+                {
+                    // Standard HTTP methods defined in RFC7231: http://tools.ietf.org/html/rfc7231#section-4.3
+                    new object[] { "GET", false },
+                    new object[] { "GET", true },
+                    new object[] { "POST", false },
+                    new object[] { "POST", true },
+                    new object[] { "PUT", false },
+                    new object[] { "PUT", true },
+                    new object[] { "DELETE", false },
+                    new object[] { "DELETE", true },
+                    
+                    // Custom HTTP method
+                    new object[] { "CUSTOM1", false },
+                    new object[] { "CUSTOM1", true }
+                };
```

Comment:
@stephentoub Thx. I knew there had to be a better way.  You're the `yield` guru!